### PR TITLE
tests/main/upgrade-from-release: correct versions for 24.04 and 24.10

### PR DESCRIPTION
**Pull request by andrewphelpsj**
This should fix `google:ubuntu-24.04-64:tests/main/upgrade-from-release`.

### CODE REVIEW

**Review by olivercalder — APPROVED** [review]
Nice, thanks! Didn't @ernestl say the deb was also released for jammy and maybe focal?

**Comment by andrewphelpsj** [comment]
@olivercalder, yeah, you're right. I didn't notice those failures though, let me see what happens here in this run.

File: tests/main/upgrade-from-release/task.yaml [hunk 1]
```diff
@@ -48,7 +48,7 @@ execute: |
     fi
 
     declare -A EXPECTED_SNAPD_VERSIONS=(
-        ["24.04"]='2.63.1\+24.04'
+        ["24.04"]='2.65.3\+24.04'
         ["22.04"]='2.55.3\+22.04'
```
Comment (ernestl) [inline tests/main/upgrade-from-release/task.yaml:52]: All of Focal, Jammy and Noble was updated today. 
See: http://archive.ubuntu.com/ubuntu/pool/main/s/snapd/

**Comment by olivercalder** [comment]
Seems focal/jammy are succeeding?

**Comment by andrewphelpsj** [comment]
Yeah let me run one of these manually and see what is up.

File: tests/main/upgrade-from-release/task.yaml [hunk 2]
```diff
@@ -49,13 +49,14 @@ execute: |
...
-        ["20.04"]='2.44.3\+20.04'
+        ["22.04"]='2.65.3\+22.04'
+        ["20.04"]='2.65.3\+20.04'
         ["18.04"]='2.32.5\+18.04'
         ["16.04"]='2.48.3'
         ["14.04"]='2.37.4~14.04.1'
     )
-    apt list --installed snapd | MATCH "${EXPECTED_SNAPD_VERSIONS[$VERSION_ID]}"
+
+    apt list --installed snapd | MATCH "${EXPECTED_SNAPD_VERSIONS[$VERSION_ID]}\]"
```
Comment (olivercalder) [inline tests/main/upgrade-from-release/task.yaml:59]: Is this a typo?

Comment (andrewphelpsj) [reply tests/main/upgrade-from-release/task.yaml:59]: No, check out the output from apt here:
```
ubuntu@noble:~$ apt list --installed snapd
Listing... Done
snapd/noble-security,now 2.63+24.04ubuntu0.1 amd64 [installed,upgradable to: 2.65.3+24.04]
```

We want to make sure we're checking on that last string, not the first one.

Comment (olivercalder) [reply tests/main/upgrade-from-release/task.yaml:59]: Ahh, nice catch. Perhaps a comment in the spread test saying as much would be useful.

Also I think there would be a problem here: what if the image already has the latest version installed, so it's
```
snapd/noble-security,now 2.65.3+24.04ubuntu0.1 amd64 [installed]
```

I think we need some sort of if-else or case check, something like:

```suggestion
    SNAPD_INSTALLED="$(apt list --installed snapd)"
    if echo "$SNAPD_INSTALLED" | MATCH 'upgradable to' ; then
        echo "$SNAPD_INSTALLED" | MATCH "${EXPECTED_SNAPD_VERSIONS[$VERSION_ID]}\]"
    else
        echo "$SNAPD_INSTALLED" | MATCH "${EXPECTED_SNAPD_VERSIONS[$VERSION_ID]}"
    fi
```

**Review by ernestl — APPROVED** [review]
Thanks!

File: tests/main/upgrade-from-release/task.yaml (hunk 2 above)
Comment (olivercalder) [reply tests/main/upgrade-from-release/task.yaml]: I think my suggestion may have been a bit naïve, you might need
```diff
- echo "$SNAPD_INSTALLED" | MATCH "upgradable to: ${EXPECTED_SNAPD_VERSIONS[$VERSION_ID]}\]$"
+ echo "$SNAPD_INSTALLED" | MATCH "upgradable to: .*${EXPECTED_SNAPD_VERSIONS[$VERSION_ID]}\]$"
```

**Comment by olivercalder** [comment]
Looks like xenial expects 2.61.4
```
 snapd/xenial-updates,xenial-security,now 2.48.3 amd64 [installed,upgradable to: 2.61.4]
```

**Comment by ernestl** [comment]
This is suppose to be the source of truth: https://launchpad.net/ubuntu/+source/snapd

**Comment by ernestl** [comment]
**This needs to be updated:**

`if os.query is-ubuntu 24.10; then
        echo "Ubuntu oracular is skipped until it is released."
        exit
fi`

@andrewphelpsj

File: tests/main/upgrade-from-release/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/main/upgrade-from-release/task.yaml:48]: thanks

**Review by sergiocazzolato — APPROVED** [review]

**Review by ernestl — APPROVED** [review]

**Pull request merged**